jetty-server/src/main/java/org/eclipse/jetty/server/handler/SizeLimitHandler.java

```diff
@@ -126,6 +126,9 @@ public boolean isOptimizedForDirectBuffers()
         @Override
         public HttpInput.Content readFrom(HttpInput.Content content)
         {
+            if (content == null)
+                return null;
```

Comment:
maybe this should also be resetting `_read` back to 0.

Reply:
I don't think so, as reading null just means there is nothing currently available, but there may be after a demand.